jcl/src/java.base/share/classes/java/lang/J9VMInternals.java

```diff
@@ -525,16 +530,23 @@ static int fastIdentityHashCode(Object anObject) {
 	static int valueHashCode(Object anObject) {
 		int hashcode;
 
+		if (null == valueObjectHashCode) {
+			try {
+				Class<?> valueObjectMethods = Class.forName("java.lang.runtime.ValueObjectMethods");
+				valueObjectHashCode = valueObjectMethods.getDeclaredMethod("valueObjectHashCode", Object.class);
```

Comment:
can we use method handle instead of reflection? method handle has a more streamlined calling sequence

Reply:
Does `MH.lookup()` hold the privilege of accessing the requested method (public or private ?) at `java.lang.runtime.ValueObjectMethods`? If so, it'd better to take a try to see how it goes.

FYI: @hangshao0

Reply:
When I try to use MH.lookup using the following code:

```java
MethodHandles.Lookup lookup = MethodHandles.lookup();
MethodType mt = MethodType.methodType(int.class, Object.class);
MethodHandle func = lookup.findStatic(valueObjectMethods, "valueObjectHashCode", mt);
```
 I get this exception:


```
java.lang.IllegalAccessException: symbolic reference class is not accessible: class java.lang.runtime.ValueObjectMethods, from class java.lang.J9VMInternals (module java.base)
	at java.base/java.lang.invoke.MemberName.makeAccessException(MemberName.java:924)
	at java.base/java.lang.invoke.MethodHandles$Lookup.checkSymbolicClass(MethodHandles.java:3736)
	at java.base/java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:3695)
	at java.base/java.lang.invoke.MethodHandles$Lookup.findStatic(MethodHandles.java:2618)
	at java.base/java.lang.J9VMInternals.valueHashCode(J9VMInternals.java:466)
	at java.base/java.lang.Object.hashCode(Object.java:123)
	at java.base/java.util.Objects.hashCode(Objects.java:105)
	at java.base/java.lang.runtime.ValueObjectMethods$MethodHandleBuilder.computeHashCode(ValueObjectMethods.java:573)
	at java.base/java.lang.runtime.ValueObjectMethods.valueObjectHashCode(ValueObjectMethods.java:801)
	at <unknown class>.<unknown method>(Unknown Source)
	... 27 more
```

Reply:
So unless I'm missing something it looks like we can't use a method handle instead